### ISSUE TITLE
ONNX Split operator not implemented

### ISSUE THREAD

**Issue by yaoyuannnn** [microsoft/onnxruntime#13811]
### Describe the issue

The ONNX Split operator gives the following error:

```
  File "/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 347, in __init__
    self._create_inference_session(providers, provider_options, disabled_optimizers)
  File "/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 395, in _create_inference_session
    sess.initialize_session(providers, provider_options, disabled_optimizers)
onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for Split(13) node with name 'Split_1003'
```

More information about the Split node:

![image](https://user-images.githubusercontent.com/5877112/205142762-359b9f57-1c01-47c0-83d6-d3b9b5c24f2c.png)
![image](https://user-images.githubusercontent.com/5877112/205142846-5cfcec62-7021-41a9-b5d8-1bd192aec93b.png)

### To reproduce

My ORT version: 1.12.0
ONNX version: 1.12.0 

### Platform

Linux

### OS Version

Ubuntu-20.04

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.12.0

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by edgchen1** [comment]
what's the data type of Split's first input?
note that only certain data types are supported:
`T = tensor(float), tensor(int32), tensor(int64), tensor(int8), tensor(string), tensor(uint8)`
https://github.com/microsoft/onnxruntime/blob/main/docs/OperatorKernels.md#cpuexecutionprovider

**Comment by yaoyuannnn** [comment]
I see. Looks like the input data type of the `Split` node is `tensor(bool)` and it's not supported yet by onnxruntime.

**Comment by edgchen1** [comment]
right. in the screenshot there's a Cast right before Split - if the input data type to the Cast is supported by Split, could try moving the Cast.

the CPU Split kernel implementation could probably be updated to handle additional data types without too much binary size increase.

**Comment by yaoyuannnn** [comment]
> the CPU Split kernel implementation could probably be updated to handle additional data types without too much binary size increase.

This will be good. In the meantime, we will try moving the Cast op.

**Comment by hippohema** [comment]
yuanbi，i am hema，mayou.i miss you。can you email me？hippohy@hotmail
